src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -269,7 +266,7 @@ where
                     unnamed_portal
                 } else {
                     // NOTE Error handle need modify later.
-                    named_portals.get_mut(&portal_name).expect("statement_name managed by client_driver, hence assume statement name always valid")
+                    named_portals.get_mut(&portal_name).expect("portal_name managed by client_driver, hence assume portal name always valid")
```

Comment:
This kind of Error handle can be modified as to return ErrorResponse instead of "expect".

This kind of Error handle cover in:
- can't find statement in 'bind' process.
- can't find portal in 'execute' process.
- can't find statement or portal  in 'describe'.  

Refer: https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-EXT-QUERY:~:text=ErrorResponse%20if%20there%20is%20no%20such%20portal.

Reply:
But I think it is strange. 
In what situation the portal_name is wrong? User specified or it's just a internal name used in client driver? If second case, then the possibilty of not found is very low and don't know why.

Reply:
I think it depends on which client driver we use.
For now, we only run on rust-postgres and you are right.
But we don't test it in other client drivers, and I'm not sure whether exist a client-driver needs users to maintain the portal name by themselves. In that way, the user should specify the portal name and it may be wrong. But this possibility sounds low.
Anyway, it's OK to maintain the current situation too?

Reply:
> it's OK to maintain the current situation too
You mean write ErrorResponse if not found like others in this PR? 

It's ok, but I was seeking a better way to code.... Otherwise we have to write a lot return Ok(true);

Reply:
Wrap process_msg()->Result<bool,Error> in do_process()?
```
do_process(){
  get_msg()
  res = process_msg(msg)
  match res{
    ...
    Error(e) => {
       write_buff(e)
    }
  }
}
```

Reply:
Yes, I have similar thoughst about this. Means we wrap every failure into each kind of Error and use `?` to jump out do_process. 

The write error response according to different kinds of Error. 

I think it can be put on the refactor roadmap.

Reply:
And we can change std::io::Error to anyhow::Error to make the life easy.